File: Chapter05/Exercise5_16/Exercise5_16.py
spark = SparkSession \
   .builder \
   .appName("exercise_sixteen") \
   .getOrCreate()

df.filter(df["salary"] > 10000).filter(df["gender"] == "Female").show()

df.filter(df.salary > 10000).filter(df.gender == "Female").show()

df.filter((df.salary > 150000) & (df.gender == "Male")).count()

df.filter((df["salary"] > 150000) | (df["gender"] == "Female")).count()

File: Chapter05-old/Exercise5_16/Exercise5_16.scala
val spark = SparkSession
   .builder()
   .appName("exercise_sixteen")
   .getOrCreate()

import spark.implicits._
df.filter($"salary" > 10000).filter($"gender" === "Female").show()

df.filter($"salary" > 150000 && $"country" === "United States").count()

df.filter($"salary" > 150000 || $"country" === "United States").count()
